src/main/java/com/auctionapp/api/repository/AuctionRepository.java

```diff
@@ -16,4 +18,19 @@
 	List<Auction> findAllByOrderByEndDateAsc();
 
 	List<Auction> findAllByCategoryId(UUID categoryId);
+
+	@Query(value = "SELECT a.id, a.start_price, a.start_date, a.end_date, a.status, a.category_id, a.item_id, a.seller_id, i.id, i.name FROM auction a, item i WHERE a.item_id = i.id AND a.id IN :auctions ORDER BY i.name ASC", nativeQuery = true)
+	List<Auction> getAuctionsSortDefault(@Param("auctions") List<UUID>  auctions);
+
+	@Query(value = "SELECT * FROM auction WHERE id IN :auctions ORDER BY end_date ASC", nativeQuery = true)
+	List<Auction> getAuctionsOldToNew(@Param("auctions") List<UUID>  auctions);
+
+	@Query(value = "SELECT * FROM auction WHERE id IN :auctions ORDER BY start_date DESC", nativeQuery = true)
+	List<Auction> getAuctionsNewToOld(@Param("auctions") List<UUID>  auctions);
+
+	@Query(value = "SELECT * FROM auction WHERE id IN :auctions ORDER BY start_price DESC", nativeQuery = true)
+	List<Auction> getAuctionsByPriceDesc(@Param("auctions") List<UUID>  auctions);
+
+	@Query(value = "SELECT * FROM auction WHERE id IN :auctions ORDER BY start_price ASC", nativeQuery = true)
+	List<Auction> getAuctionsByPriceAsc(@Param("auctions") List<UUID> auctions);
```

Comment:
I'm not sure these should be repository methods, since you already have the auctions you want to sort (I guess since you clearly have their ids? Where are those coming from?). If the reason you have these is that you are applying multiple filters, ideally you would build the implementation so that you have a single API that can accept any kind of filter you'd like to apply, and then if present, you'd apply it and get the appropriate results back.